tutorials/inner_products/1_basic.py

```diff
@@ -0,0 +1,272 @@
+"""
+Basic Inner Products
+====================
+
+Inner products between two scalar or vector quantities represents the most
+basic class of inner products. For this class of inner products, we demonstrate:
+
+    - How to construct the inner product matrix
+    - How to use inner product matricies to approximate the inner product
+    - How to construct the inverse of the inner product matrix.
+
+For scalar quantities :math:`\\psi` and :math:`\\phi`, the
+inner product is given by:
+
+.. math::
+    (\\psi , \\phi ) = \\int_\\Omega \\psi \\, \\phi \\, dv
+
+
+And for vector quantities :math:`\\vec{u}` and :math:`\\vec{v}`, the
+inner product is given by:
+
+.. math::
+    (\\vec{u}, \\vec{v}) = \\int_\\Omega \\vec{u} \\cdot \\vec{v} \\, dv
+
+
+In discretized form, we can approximate the aforementioned inner-products as:
+
+.. math::
+    (\\psi , \\phi) \\approx \\mathbf{\\psi^T \\, M \\, \\phi}
+
+
+and
+
+.. math::
+    (\\vec{u}, \\vec{v}) \\approx \\mathbf{u^T \\, M \\, v}
+
+
+where :math:`\\mathbf{M}` in either equation represents an
+*inner-product matrix*. :math:`\\mathbf{\\psi}`, :math:`\\mathbf{\\phi}`,
+:math:`\\mathbf{u}` and :math:`\\mathbf{v}` are discrete variables that live
+on the mesh. It is important to note a few things about the
+inner-product matrix in this case:
+
+    1. It depends on the dimensions and discretization of the mesh
+    2. It depends on where the discrete variables live; e.g. edges, faces, nodes, centers
+
+For this simple class of inner products, the inner product matricies for
+discrete quantities living on varioud parts of the mesh have the form:
+
+.. math::
+    \\textrm{Centers:} \\; \\mathbf{M_c} &= \\textrm{diag} (\\mathbf{v} ) \n
+    \\textrm{Nodes:} \\; \\mathbf{M_n} &= \\frac{1}{2^{2k}} \\mathbf{P_n^T } \\textrm{diag} (\\mathbf{v} ) \\mathbf{P_n} \n
+    \\textrm{Faces:} \\; \\mathbf{M_f} &= \\frac{1}{4} \\mathbf{P_f^T } \\textrm{diag} (\\mathbf{I_k \\otimes v} ) \\mathbf{P_f} \n
+    \\textrm{Edges:} \\; \\mathbf{M_e} &= \\frac{1}{4^{k-1}} \\mathbf{P_e^T } \\textrm{diag} (\\mathbf{I_k \\otimes v}) \\mathbf{P_e}
+
+where :math:`k = 1,2,3`, :math:`\\mathbf{I_k}` is the identity matrix and
+:math:`\\otimes` is the kronecker product. :math:`\\mathbf{P}` are projection
+matricies that map quantities from one part of the cell (nodes, faces, edges)
+to cell centers.
+    
+
+"""
+
+####################################################
+#
+# Import Packages
+# ---------------
+#
+# Here we import the packages required for this tutorial
+#
+
+from discretize.utils.matutils import sdiag
```

Comment:
```suggestion
from discretize.utils import sdiag
```

This simplifies the import a bit (and also eases thing if we move the files where sdiag is within the utils)